src/pke/lib/scheme/ckksrns/ckksrns-parametergeneration.cpp

```diff
@@ -111,9 +115,11 @@ bool ParameterGenerationCKKSRNS::ParamsGenCKKSRNS(std::shared_ptr<CryptoParamete
 
     //// HE Standards compliance logic/check
     SecurityLevel stdLevel = cryptoParamsCKKSRNS->GetStdLevel();
-    uint32_t auxBits       = (scalTech == COMPOSITESCALINGAUTO || scalTech == COMPOSITESCALINGMANUAL) ? 30 : AUXMODSIZE;
-    // Alternative way to calculate auxBits for composite scaling mode
-    //                         ((registerWordSize < 30) ? registerWordSize : 30) : AUXMODSIZE;
+    uint32_t auxBits =
+        ((scalTech == COMPOSITESCALINGAUTO || scalTech == COMPOSITESCALINGMANUAL) && registerWordSize <= AUXMODSIZE) ?
+            registerWordSize -
+                1 :  // Duhyeong: Let's check if auxBits = registerWordSize makes an error in the P prime generation.
```

Comment:
does this line require an additional review and test?

Reply:
I have reviewed it already. This simplifies things and keep the LogP estimate procedure as the original implementation. This also guarantees secure parameter generation, on which @du1204 can elaborate more since he is part of the HE standard committee.